hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -264,50 +261,71 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(final Lon
-            entityManager.detach(distributionSetEntity);
-            // return with nothing as all targets had the DS already assigned
-            return new DistributionSetAssignmentResult(distributionSetEntity, Collections.emptyList(), 0,
-                    targetsWithActionType.size(), Collections.emptyList(), targetManagement);
+            return allTargetsAlreadyAssignedResult(distributionSetEntity, targetsWithActionType.size());
         }
 
-        // split tIDs length into max entries in-statement because many database
-        // have constraint of max entries in in-statements e.g. Oracle with
-        // maximum 1000 elements, so we need to split the entries here and
-        // execute multiple statements
-        final List<List<Long>> targetEntitiesIdsChunks = Lists.partition(
-                targetEntities.stream().map(Target::getId).collect(Collectors.toList()),
-                Constants.MAX_ENTRIES_IN_STATEMENT);
+        final Map<String, JpaAction> assignedActions = doAssignDistributionSetToTargets(targetsWithActionType,
+                actionMessage, assignmentStrategy, distributionSetEntity, targetEntities);
+        return buildAssignmentResult(distributionSetEntity, assignedActions, targetsWithActionType);
+    }
+
+    private DistributionSetAssignmentResult allTargetsAlreadyAssignedResult(
+            final JpaDistributionSet distributionSetEntity, final int alreadyAssignedCount) {
+        // detaching as it is not necessary to persist the set itself
+        entityManager.detach(distributionSetEntity);
+        // return with nothing as all targets had the DS already assigned
+        return new DistributionSetAssignmentResult(distributionSetEntity, alreadyAssignedCount,
+                Collections.emptyList());
+    }
 
+    private Map<String, JpaAction> doAssignDistributionSetToTargets(
+            final Collection<TargetWithActionType> targetsWithActionType, final String actionMessage,
+            final AbstractDsAssignmentStrategy assignmentStrategy, final JpaDistributionSet distributionSetEntity,
+            final List<JpaTarget> targetEntities) {
+        final List<List<Long>> targetEntitiesIdsChunks = getTargetEntitiesAsChunks(targetEntities);
         closeOrCancelActiveActions(assignmentStrategy, targetEntitiesIdsChunks);
         // cancel all scheduled actions which are in-active, these actions were
         // not active before and the manual assignment which has been done
         // cancels them
         targetEntitiesIdsChunks.forEach(this::cancelInactiveScheduledActionsForTargets);
-
         setAssignedDistributionSetAndTargetUpdateStatus(assignmentStrategy, distributionSetEntity,
                 targetEntitiesIdsChunks);
-
-        final Map<String, JpaAction> controllerIdsToActions = createActions(targetsWithActionType, targetEntities,
+        final Map<String, JpaAction> assignedActions = createActions(targetsWithActionType, targetEntities,
```

Comment:
Please check if it is needed to store a Map here, or if a collection of JpaAction would be enough. (The key seems never to be used, probably createActions can be adapted to return the actions only )

Reply:
good catch